app/src/main/java/com/rob729/quiethours/Fragments/MainFragment.kt

```diff
@@ -82,7 +82,16 @@ class MainFragment : Fragment() {
             } else {
                 binding.emrl.visibility = View.GONE
             }
-
+            var i = 0
```

Comment:
change "profileViewModel.allProfiles.observe(this, Observer {" to "profileViewModel.allProfiles.observe(this, Observer { profilesList ->"
and in place of while loop placed below used profilesList.forEach{}.
If Kotlin is providing some functionality for making code clean then try to use those features.

Reply:
@rob729 Done